python_modules/libraries/dagstermill/dagstermill_tests/test_assets.py

```diff
@@ -0,0 +1,548 @@
+import importlib.util
+import os
+import pickle
+import tempfile
+from contextlib import contextmanager
+
+import nbformat
+import pytest
+from dagstermill import DagstermillError
+from dagstermill.compat import ExecutionError
+from dagstermill.factory import define_dagstermill_solid
+from dagstermill.examples.repository import notebook_assets_repo
+from jupyter_client.kernelspec import NoSuchKernel
+from nbconvert.preprocessors import ExecutePreprocessor
+
+from dagster._check import CheckError
+from dagster._core.definitions.materialize import materialize
+from dagster._core.definitions.metadata import NotebookMetadataValue
+from dagster._core.definitions.reconstruct import ReconstructablePipeline
+from dagster._core.test_utils import instance_for_test
+from dagster._legacy import execute_pipeline, pipeline
+from dagster._utils import file_relative_path, safe_tempfile_path
+from dagster import AssetKey, reconstructable
+
+DAGSTER_PANDAS_PRESENT = importlib.util.find_spec("dagster_pandas") is not None
+SKLEARN_PRESENT = importlib.util.find_spec("sklearn") is not None
+MATPLOTLIB_PRESENT = importlib.util.find_spec("matplotlib") is not None
+
+
+def get_path(materialization_event):
+    for (
+        metadata_entry
+    ) in materialization_event.event_specific_data.materialization.metadata_entries:
+        if isinstance(metadata_entry.entry_data, NotebookMetadataValue):
+            return metadata_entry.entry_data.path
+
+
+def cleanup_result_notebook(result):
+    if not result:
+        return
+    materialization_events = [
+        x for x in result.step_event_list if x.event_type_value == "ASSET_MATERIALIZATION"
+    ]
+    for materialization_event in materialization_events:
+        result_path = get_path(materialization_event)
+        if os.path.exists(result_path):
+            os.unlink(result_path)
+
+
+@contextmanager
+def exec_for_test(asset_name, env=None, raise_on_error=True, **kwargs):
+    result = None
+    # assets = [notebook_assets_repo._assets_defs_by_key.get(key) for key in asset_keys]
+    recon_pipeline = ReconstructablePipeline.for_module("dagstermill.examples.repository", asset_name)
```

Comment:
is this where the test is getting errors?

Reply:
yeah, i can't figure out how to materialize the assets. i tried just doing a plain materialize, converting the assets to jobs with build_asset_job and a couple other things i dont remember and nothing worked